src/bootstrap/build/step.rs

```diff
@@ -283,7 +290,8 @@ impl<'a> Step<'a> {
                      self.doc_error_index(stage)]
             }
             Source::Check { stage, compiler: _ } => {
-                vec![self.check_linkcheck(stage)]
+                vec![self.check_linkcheck(stage),
+                     self.dist(stage)]
```

Comment:
Note that for now I've modified `make check` to run `make dist` as well essentially, should help us ensure that code doesn't break *too* badly.